packages/app/src/cli/models/extensions/specification.ts

```diff
@@ -175,6 +202,25 @@ export class UIExtensionInstance<TConfiguration extends BaseConfigContents = Bas
   hasExtensionPointTarget(target: string): boolean {
     return this.specification.hasExtensionPointTarget?.(this.configuration, target) || false
   }
+
+  // Functions specific properties
+  get buildCommand() {
+    const config = this.configuration as unknown as FunctionSchemaContents
```

Comment:
If `this.configuration` isn't a Function-style one, then presumably at runtime this method (and the others) would fail? (and in a similar fashion, the UI specific properties above)

Is there more we can do to make using an `ExtensionInstance` as safe as possible?